enterprise/server/execution_service/execution_service.go

```diff
@@ -40,9 +40,16 @@ func checkPreconditions(req *espb.GetExecutionRequest) error {
 func (es *ExecutionService) queryExecutions(ctx context.Context, baseQuery *query_builder.Query) ([]tables.Execution, error) {
 	dbh := es.env.GetDBHandle()
 	q := baseQuery
-	if err := perms.AddPermissionsCheckToQueryWithTableAlias(ctx, es.env, q, "e"); err != nil {
+
+	permClauses, err := perms.GetPermissionsCheckClauses(ctx, es.env, q, "e")
+	if err != nil {
 		return nil, err
 	}
+	// If the invocation is public, allow viewing its execution metadata.
```

Comment:
could you add a more detailed comment explaining why this is necessary?

Reply:
Done